packages/kbn-search-connectors/lib/update_connector_scheduling.ts

```diff
@@ -7,35 +7,40 @@
  */
 
 import { ElasticsearchClient } from '@kbn/core/server';
+import { errors } from '@elastic/elasticsearch';
 import { i18n } from '@kbn/i18n';
 
-import { CONNECTORS_INDEX } from '..';
-
-import { ConnectorDocument, SchedulingConfiguraton } from '../types/connectors';
+import { SchedulingConfiguraton } from '../types/connectors';
+import { ConnectorsAPIUpdateResponse } from '../types/connectors_api';
 
 export const updateConnectorScheduling = async (
   client: ElasticsearchClient,
   connectorId: string,
   scheduling: SchedulingConfiguraton
 ) => {
-  const connectorResult = await client.get<ConnectorDocument>({
-    id: connectorId,
-    index: CONNECTORS_INDEX,
-  });
-  const connector = connectorResult._source;
-  if (connector) {
-    const result = await client.index<ConnectorDocument>({
-      document: { ...connector, scheduling },
-      id: connectorId,
-      index: CONNECTORS_INDEX,
+  try {
+    const result = await client.transport.request<ConnectorsAPIUpdateResponse>({
+      method: 'PUT',
+      path: `/_connector/${connectorId}/_scheduling`,
+      body: {
+        scheduling,
+      },
     });
-    await client.indices.refresh({ index: CONNECTORS_INDEX });
     return result;
-  } else {
-    throw new Error(
-      i18n.translate('searchConnectors.server.connectors.scheduling.error', {
-        defaultMessage: 'Could not find document',
-      })
-    );
+  } catch (err) {
+    const isResponseError = err instanceof errors.ResponseError;
+
+    const isDocumentMissingError =
```

Comment:
We should move this to https://github.com/elastic/kibana/tree/main/packages/kbn-search-connectors/utils/identify_exceptions.ts so that we can re-use it in other endpoints.